config/routes.rb

```diff
@@ -3,7 +3,8 @@
 
   root to: "home#index"
 
-  resources :users do
+  resources :teachers, except: [:index] do
+    get "/students", to: "teachers#show_students", as: "students"
```

Comment:
`resources :students, only: %i[show]`

try using something like this pls, so as I mentioned before students #show action should be in its own controller

Reply:
fixed